taca/deliver/deliver.py

```diff
@@ -0,0 +1,385 @@
+"""
+    Module for controlling deliveries of samples and projects
+"""
+import datetime
+import glob
+import hashlib
+import os
+import re
+from ngi_pipeline.database import classes as db
+from ngi_pipeline.utils.classes import memoized
+from taca.log import LOG
+from taca.utils.config import CONFIG
+from taca.utils.filesystem import create_folder
+from taca.utils.misc import hashfile
+from taca.utils import transfer
+
+class DelivererError(Exception): pass
+class DelivererDatabaseError(DelivererError): pass
+class DelivererReplaceError(DelivererError): pass
+class DelivererRsyncError(DelivererError): pass
+
+class Deliverer(object):
+    """ 
+        A (abstract) superclass with functionality for handling deliveries
+    """
+    def __init__(self, projectid, sampleid, **kwargs):
+        """
+            :param string projectid: id of project to deliver
+            :param string sampleid: id of sample to deliver
+            :param bool no_checksum: if True, skip the checksum computation
+            :param string hash_algorithm: algorithm to use for calculating 
+                file checksums, defaults to sha1
+        """
+        self.log = LOG
+        # override configuration options with options given on the command line
+        self.config = CONFIG.get('deliver',{})
+        self.config.update(kwargs)
+        # set items in the configuration as attributes
+        for k,v in self.config.items():
+            setattr(self,k,v)
```

Comment:
well, I'm sure this violates a lot of good-practices-principles and I may refactor it